pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -116,6 +123,26 @@ type BGPStandardCommunity string
 // +kubebuilder:validation:Pattern=`^([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5]):([0-9]|[1-9][0-9]{1,8}|[1-3][0-9]{9}|4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|429496[0-6][0-9]{3}|4294967[01][0-9]{2}|42949672[0-8][0-9]|429496729[0-5])$`
 type BGPLargeCommunity string
 
+// BGPExtendedCommunity type represents a value of the BGP Extended Communities Attribute (RFC 4360).
+type BGPExtendedCommunity struct {
+	// SubType indicates a particular sub-type of extended community.
+	//
+	// +kubebuilder:validation:Enum=RouteTarget;RouteOrigin;LinkBandwidth
+	// +kubebuilder:validation:Required
+	SubType string `json:"subType"`
```

Comment:
Since we are defining the constants BGPExtendedCommunityRouteTarget, should we have enum type for SubType ?

```
// +kubebuilder:validation:Enum=RouteTarget;RouteOrigin;LinkBandwidth
type BGPExtendedCommunityType string

const (
BGPExtendedCommunityRouteTarget BGPExtendedCommunityType = "RouteTarget"
...
)
...
type BGPExtendedCommunity struct {
   SubType BGPExtendedCommunityType
... 
}
```